packages/loader/container-utils/src/error.ts

```diff
@@ -48,6 +49,48 @@ export class DataCorruptionError extends LoggingError implements IErrorBase {
     }
 }
 
+/**
+ * Coerce the throwable input into a DataCorruptionError.
+ * @param error - Throwable input to be converted.
+ */
+export function CreateCorruptionError(error: any): DataCorruptionError {
+    if (typeof error === "string") {
+        return new DataCorruptionError(error, {});
+    } else if (!error || Array.isArray(error) || typeof error !== "object") {
+        return new DataCorruptionError(
+            "DataCorruptionError without explicit message (needs review)",
+            { typeof: typeof error },
+        );
+    } else if (error instanceof DataCorruptionError) {
+        return error;
+    } else if (error instanceof LoggingError) {
+        const { errorType: errorSubType } = error as any;
+
+        return new DataCorruptionError(
```

Comment:
I'm wondering if we want to not coerce other logging error types, so code has a way to not log a corruption error. otherwise there is no way not to log a corruption error

Reply:
I don't (yet) have enough context to weigh in... but I might ask the following to generate food for thought:
* When are non-LoggingErrors thrown (and where are they coming from)?
* When are LoggingErrors thrown (and where are they coming from)?
* Should the behavior differ based on callsite
  * [deltaManager.ts](https://github.com/microsoft/FluidFramework/pull/5144/files#diff-01eeb7567dd98442afb9a308001150972f600ef3f04e17a71f88b34a0b8ee153R403)
  * [channelDeltaConnection.ts](https://github.com/microsoft/FluidFramework/pull/5144/files#diff-27ae7b1e16da168c508c887daa4cc945211fd4e7e2ccb0bf4f4973edb3c43a0aR43)
  * [deltaStoreRuntime.ts](https://github.com/microsoft/FluidFramework/pull/5144/files#diff-5b145dfa0151b5be1f17d42f73c3aa248d3600b934665f3d22311345d32df4e4R587)

Reply:
most layers don't use logging errors. we can always adjust later

Reply:
Related to the [comment below](https://github.com/microsoft/FluidFramework/pull/5144#pullrequestreview-591623644) perhaps [deltaManager.ts](https://github.com/microsoft/FluidFramework/pull/5144/files#diff-01eeb7567dd98442afb9a308001150972f600ef3f04e17a71f88b34a0b8ee153R403) should not coerce to DataCorruptionError at all since it does not seem to have access to a lot of the related info we'd probably want anyway(?)